wavedump/src/analyze-waveforms

```diff
@@ -2,41 +2,12 @@
 from __future__ import print_function, division
 import h5py
 import numpy as np
+import pandas as pd
+from scipy import signal
 
 def iqr(x):
     return np.percentile(x,75) - np.percentile(x,25)
 
-def get_threshold_crossing(x, data, threshold=0.4, rising=True):
-    """
-    Returns the times at which the waveforms in `x` cross 40% of their minimum
-    value.
-    """
-    data = np.asarray(data)
-    argmin = np.argmin(data,axis=-1)
-    thresholds = threshold*data[np.arange(data.shape[0]),argmin]
-    if rising:
-        il = data.shape[1]-np.argmax(((np.arange(data.shape[1]) < argmin[:,np.newaxis]) & (data > thresholds[:,np.newaxis]))[:,::-1],axis=-1)-1
-        ir = il + 1
-        ir[ir >= data.shape[1]] = data.shape[1]-1
-        i = np.arange(data.shape[0])
-        return x[il] + (thresholds-data[i,il])*(x[ir]-x[il])/(data[i,ir]-data[i,il])
-    else:
-        il = data.shape[1]-np.argmax(((np.arange(data.shape[1]) > argmin[:,np.newaxis]) & (data < thresholds[:,np.newaxis]))[:,::-1],axis=-1)-1
-        ir = il + 1
-        ir[ir >= data.shape[1]] = data.shape[1]-1
-        i = np.arange(data.shape[0])
-        return x[il] + (thresholds-data[i,il])*(x[ir]-x[il])/(data[i,ir]-data[i,il])
-
-def get_rise_time(x, data):
-    t10 = get_threshold_crossing(x, data, 0.1)
-    t90 = get_threshold_crossing(x, data, 0.9)
-    return t90 - t10
-
-def get_fall_time(x, data):
-    t10 = get_threshold_crossing(x, data, 0.1, rising=False)
-    t90 = get_threshold_crossing(x, data, 0.9, rising=False)
-    return t10 - t90
-
```

Comment:
I think these functions could be useful in the future. Actually, it looks like get_threshold_crossing() and get_times() are pretty similar, so those should just be one function. The nice thing about get_threshold_crossing is that it has the ability to specify the threshold and whether you are looking for a rising or falling pulse, so maybe we should stick with that one and then update the code elsewhere to call that. For the rise and fall times, currently we are not really using this information but it could be useful in the future.

Reply:
I see what you mean. I actually didn't know these methods existed before doing the pull request because they weren't in the acquisition repository. Instead, there was `get_window`, which I thought was newer, but I'll work on implementing `get_threshold_crossing`.